oximeter/collector/src/lib.rs

```diff
@@ -258,35 +240,55 @@ impl Oximeter {
                 .map(|ip| SocketAddr::new(ip, DNS_PORT))
                 .collect();
 
-        let make_clickhouse_resolver = || -> BoxedResolver {
+        // Closure to create _two_ resolvers, one to resolve the ClickHouse HTTP
+        // SRV record, and one for the native TCP record.
+        //
+        // TODO(cleanup): This should be removed if / when we entirely switch to
+        // the native protocol.
+        let make_clickhouse_resolvers = || -> (BoxedResolver, BoxedResolver) {
             if let Some(address) = config.db.address {
-                Box::new(SingleHostResolver::new(address))
+                let http = Box::new(SingleHostResolver::new(address));
+                let native_addr =
+                    SocketAddr::new(address.ip(), CLICKHOUSE_TCP_PORT);
```

Comment:
Should this use `config.db.native_address` if it was provided?

Reply:
Yeah, good catch. I rework this to separately make the two resolvers, thanks.

Reply:
Done in 5e96ce062